app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -4,39 +4,65 @@
 
 package org.mozilla.fenix.settings.creditcards
 
+import android.app.KeyguardManager
+import android.content.Context
+import android.content.DialogInterface
+import android.content.Intent
 import android.os.Bundle
+import android.provider.Settings
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AlertDialog
 import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
 import androidx.preference.Preference
-import androidx.preference.PreferenceFragmentCompat
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import mozilla.components.concept.storage.CreditCard
 import mozilla.components.lib.state.ext.consumeFrom
 import mozilla.components.service.fxa.SyncEngine
+import org.mozilla.fenix.NavGraphDirections
 import org.mozilla.fenix.R
 import org.mozilla.fenix.components.StoreProvider
 import org.mozilla.fenix.ext.getPreferenceKey
 import org.mozilla.fenix.ext.navigateBlockingForAsyncNavGraph
 import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.ext.runIfFragmentIsAttached
+import org.mozilla.fenix.ext.secure
 import org.mozilla.fenix.ext.showToolbar
 import org.mozilla.fenix.settings.SyncPreferenceView
+import org.mozilla.fenix.settings.biometric.BiometricPromptPreferenceFragment
 import org.mozilla.fenix.settings.requirePreference
 
 /**
  * "Credit cards" settings fragment displays a list of settings related to autofilling, adding and
- * syncing credit cards.
+ * syncing credit cards. Authentication for saved credit cards using [BiometricPromptFeature]
+ * or [KeyguardManager] (similar to [SavedLoginsAuthFragment]).
  */
-class CreditCardsSettingFragment : PreferenceFragmentCompat() {
+@SuppressWarnings("TooManyFunctions")
+class CreditCardsSettingFragment : BiometricPromptPreferenceFragment() {
 
```

Comment:
Question: Do we really need to secure this fragment? Is there sensitive data displayed here? Fo screenshots we haven't blocked this screen from being screenshotted.

Reply:
This is following the same flow that we have for logins (`SavedLoginsAuthFragment`) where we ask the user to authenticate before they can see their saved/synced info. We need to talk to UX about whether this flow should be the same or changed (see Grisha's comment above)